src/views/explore/PhenotypeExplorer.vue

```diff
@@ -0,0 +1,282 @@
+    <li>Paste comma-separated phenotype IDs</li>
+  </ol>`;
+
+const bModeOptions = [
+  "phenotypes from all genes of ...",
+  "phenotypes from all human diseases",
+  "these phenotypes ...",
+];
+const bTaxonOptions = ["mouse", "zebrafish", "fruitfly", "nematode", "frog"];
+
+const exampleAPhenotypes = [
+  { value: "HP:0004970" },
+  { value: "HP:0004933" },
+  { value: "HP:0004927" },
+  { value: "HP:0002108" },
+  { value: "HP:0004872" },
+  { value: "HP:0012499" },
+  { value: "HP:0002650" },
+];
+
+const exampleBPhenotypes = [
+  { value: "FBcv:0000366" },
+  { value: "FBcv:0000439" },
+  { value: "FBcv:0002041" },
+  { value: "FBcv:0002039" },
+];
+
+// phenotype explore mode that compares two sets of phenotypes
+export default defineComponent({
+  components: {
+    AppSelectTags,
+    AppSelectSingle,
+    AppStatus,
+    AppRing,
+  },
+  data() {
+    return {
+      // first set of phenotypes
+      aPhenotypes: [] as Options,
+      // options for mode of second set
+      bModeOptions,
+      // selected mode of second set
+      bMode: bModeOptions[0],
+      // taxon options for second set
+      bTaxonOptions,
+      // selected taxon for second set
+      bTaxon: bTaxonOptions[0],
+      // second set of phenotypes
+      bPhenotypes: [] as Options,
+      // common tooltip for multi-select component
+      multiTooltip,
+      // status of analysis
+      status: null as Status | null,
+      // analysis results
+      results: { matches: [] } as Results,
+    };
+  },
+  methods: {
+    // example phenotype set comparison
+    doExample() {
+      this.aPhenotypes = exampleAPhenotypes;
+      this.bPhenotypes = exampleBPhenotypes;
+      this.bMode = bModeOptions[2];
+    },
+    // run comparison analysis
+    async runAnalysis() {
+      // loading...
+      this.status = { code: "loading", text: "Analyzing phenotypes" };
+      this.results = { matches: [] };
+
+      try {
+        // run appropriate analysis based on selected mode
+        if (this.bMode.includes("phenotypes from"))
+          this.results = await compareSetToGene(
+            this.aPhenotypes.map(({ value }) => String(value)),
+            this.bMode.includes("diseases") ? "human" : this.bTaxon
+          );
+        else
+          this.results = await compareSetToSet(
+            this.aPhenotypes.map(({ value }) => String(value)),
+            this.bPhenotypes.map(({ value }) => String(value))
+          );
+
+        // clear status
+        this.status = null;
+      } catch (error) {
+        // error...
+        this.status = error as ApiError;
+        this.results = { matches: [] };
+      }
+    },
+    // when multi select component runs value function
+    valueFunc(length: number) {
+      if (length === 0) push("No associated phenotypes found");
+      else push(`Selected ${length} phenotypes`);
+    },
+    // show results in phenogrid
+    phenogrid(id: string) {
```

Comment:
Just checking, but I assume this will eventually load up a Phenogrid instance comparing against just this row's contents? This behavior seems a bit different than the 2.0 version, where the Phenogrid instance is displayed for the entire query's results, versus being displayed for individual rows in the query results. I admit that I'm ignorant of what input Phenogrid accepts, so maybe this point doesn't make sense.

Reply:
From playing around with Phenogrid a bit in 2.0, it seems to take two sets of phenotypes, then displays a matrix where each phenotype is compared to every other one. Does that mean you'll usually have a set of phenotypes in each rows of this results table, then?

Reply:
The "top 10" results shown in this PR is more equivalent to this part of the 2.0 UI:

<img width="1416" alt="image" src="https://user-images.githubusercontent.com/8326331/159045695-cd1bd95f-d39e-4e4c-9f78-4695fd587f1c.png">

Except that we'll only be showing top 10, instead of a table, and possibly adding a continuous "show more" button below it. This was decided upon in a design discussion with Tim and others, trying to make the results more palatable. The list is supposed to just mirror the list of "match results" returned from the endpoint.

The next PR will incorporate actual Phenogrid results (if I can get ever phenogrid to properly run in this repo; I'm working on it), which shows the matrix of either `aPhenotypes` vs `bTaxon` (genes of certain species, or human disease) or `aPhenotypes` vs `bPhenotypes`.

Does that address your question?

Reply:
Sort of; sorry, I wasn't clear with my original question. I was wondering about the "See Details" button that I assume is going to render in each row -- based on the name of this method and the fact that it takes a single ID, I presumed it'd load Phenogrid with just one comparison ID (i.e., the ID from the row in which you clicked "See Details"). That doesn't match my current understanding of Phenogrid, i.e. that it compares one set of phenotypes to another set, not to a single ID. If you actually intend to display Phenogrid for the entire query and the "See Details" button is going to do something else, then disregard this comment.

FWIW, everything else you said about the top 10 list sounds good to me.

Reply:
Ahh okay sorry. The "see details" button was actually a not-well-formed idea by me. I thought it would be nice to have some way to "highlight" the match in the phenogrid viz below it, as a way to "drill down" to get more information.

However, looking through the phenogrid docs, it doesn't seem like such a thing is even possible anyway (doesn't seem like you can programmatically control which matrix cell is highlighted, for example). So I suppose we just get rid of this for now.

Alternatively, it could show just a drop down below the entry with more metadata returned from the `/sim` endpoint. But looking at that metadata, I'm not sure how much is meaningful/useful to the user beyond what I've already included (the label and the id).

Reply:
Ah, ok, that makes a lot more sense. Then yeah, maybe it's best to comment out the button until we know what kind of data we'd display on a per-row basis.

That said, I do like the idea of being able to highlight a row/column in Phenogrid's matrix, though, as the Phenogrid labels are not super informative. For example, if you want to figure out which ID in your query corresponds to a row/column, you have to hover over the label, then inspect the URL of the link to see the HPO ID. It'd be nice if there were a clearer connection between the things you queried for and Phenogrid's row/column labels. I don't know how much time we have to muck around in Phenogrid's implementation, but if other people think it's a good idea I could take a crack at it.

Reply:
I've also deleted the button for now in the next PR as well, if that's okay.

You could certainly take a stab if you have time and want to, but that package has so many problems i'd honestly rather wait to just rewrite it. We could even use new features as an incentive for us to do a rewrite.

Reply:
Agreed; it'd be nice to get some leverage to rewrite it aside from it just being a pain for us devs to work with it 🙂 .